src/data.rs

```diff
@@ -61,7 +61,7 @@ where
     ) -> BoxedFuture<'a, Result<Self::Asset, Self::Error>> {
         Box::pin(async move {
             let mut bytes = Vec::new();
-            reader.read_to_end(&mut bytes).await.unwrap();
+            reader.read_to_end(&mut bytes).await.map_err(CustomJsonLoaderError::Io).unwrap();
```

Comment:
The purpose of mapping this error is to propagate it.

```suggestion
            reader.read_to_end(&mut bytes).await.map_err(CustomJsonLoaderError::Io)?;
```